src/Umbraco.Web/Routing/IMediaUrlProvider.cs

```diff
@@ -26,6 +25,6 @@ public interface IMediaUrlProvider
         /// e.g. a cdn url provider will most likely always return an absolute url.</para>
         /// <para>If the provider is unable to provide a url, it returns <c>null</c>.</para>
         /// </remarks>
-        UrlInfo GetMediaUrl(IUmbracoContext umbracoContext, IPublishedContent content, string propertyAlias, UrlMode mode, string culture, Uri current);
```

Comment:
Seems like by changing this signature we've invented new circular dependencies?

Reply:
This is the cycle..  
```
DefaultMediaUrlProvider
- PropertyEditorCollection
-- DataEditorCollection
--- GridPropertyEditor
---- HtmlImageSourceParser
----- UrlProvider
------ MediaUrlProviderCollection
------- DefaultMediaUrlProvider
```

I can't find a good way to break it currently.  I think `DefaultMediaUrlProvider` - where it is currently solved by having a lazy dependency is the correct place to break the cycle.